lib/ui/src/Cardano/Wallet/UI/Lib/Address.hs

```diff
@@ -0,0 +1,29 @@
+{-# LANGUAGE QuasiQuotes #-}
+module Cardano.Wallet.UI.Lib.Address
+where
+
+import Prelude
+
+import Cardano.Ledger.Address
+    ( unCompactAddr
+    )
+import Cardano.Wallet.Deposit.Read
+    ( Address
+    )
+import Codec.Binary.Bech32
+    ( dataPartFromBytes
+    )
+import Data.Text
+    ( Text
+    )
+
+import qualified Codec.Binary.Bech32 as Bech32
+import qualified Codec.Binary.Bech32.TH as Bech32
+import qualified Data.ByteString.Short as B8
+
+encodeMainnetAddress :: Address -> Text
+encodeMainnetAddress addr = bech32
+  where
+    bytes = B8.fromShort $ unCompactAddr addr
+    bech32 = Bech32.encodeLenient hrp (dataPartFromBytes bytes)
+    hrp = [Bech32.humanReadablePart|addr|]
```

Comment:
I have zero idea what static guarantees the authors of `Bech32` expected to gain through this Template Haskell contraption. 🙈 Thank you for isolating it in a single module. 🙏